parser/parser.coffee

```diff
@@ -0,0 +1,118 @@
+scanner = require '../scanner/scanner'
+CustomError = require '../error/custom_error'
+
+Program = require '../entities/01_program'
+Block = require '../entities/02_block'
+ForStatement = require '../entities/03_for_statement'
+WhileStatement = require '../entities/03_while_statement'
+
+StartTokens = require './start_tokens'
+
+tokens = []
```

Comment:
So our scanner is a class and does not have any variables that are global to the module. For consistency, we should decide if we want to have classes or just have variables global to each module.